src/plopp/graphics/color_mapper.py

```diff
@@ -40,52 +45,111 @@ def __init__(self,
         self.mask_cmap = _get_cmap(mask_cmap, nan_color=nan_color)
         self.user_vmin = vmin
         self.user_vmax = vmax
-        self.vmin = np.inf
-        self.vmax = np.NINF
-        self.norm_flag = norm
-        self.norm_func = None
+        self._vmin = {'linear': np.inf, 'log': np.inf}
+        self._vmax = {'linear': np.NINF, 'log': np.NINF}
+        self._norm = norm
+        self.normalizer = {'linear': Normalize(), 'log': LogNorm()}
+        self._notify_on_change = [notify_on_change]
+        self.colorbar = None
+        self.unit = None
+        self.name = None
 
-    def rescale(self, data):
+    @property
+    def widget(self):
+        import ipywidgets as ipw
+        self.colorbar = {
+            'image':
+            ipw.Image(),
+            'button':
+            ToggleTool(self.toggle_norm,
+                       value=self._norm == 'log',
+                       description='log',
+                       tooltip='Toggle data norm').widget
+        }
+        self._update_colorbar()
+        return ipw.VBox(list(self.colorbar.values()))
+
+    def _update_colorbar(self):
+        dpi = 96
+        height_px = 600
```

Comment:
Should this be tied to the actual height of the axes that it is attached to?

Reply:
Because the `ColorMapper` is not directly aware of the figure it is used in, it was quite messy to pass around the figure size everywhere. So instead I picked an average size, and then I resize it here: https://github.com/scipp/plopp/blob/slicing_in_3d/src/plopp/graphics/scatterfig.py#L106
(with some scaling factor to allow some space for the `log` button)

It would look cleaner if we used the real size of the figure directly, as it doesn't always look super when it scales the image down or up, but I went for code simplicity instead of looks. We can change.

Reply:
It is fine for now. We can do this kind of polishing later when we have a more or less finished layout.